File: examples/snippets/modules/ROOT/examples/pekkostreams-java/src/main/java/cloudflow/pekkostreamsdoc/DataMerge.java
package cloudflow.pekkostreamsdoc;

// tag::merge[]
import cloudflow.pekkostream.*;
import cloudflow.streamlets.*;
import cloudflow.streamlets.avro.*;
import cloudflow.pekkostream.javadsl.*;
import cloudflow.pekkostream.util.javadsl.*;
import org.apache.pekko.stream.javadsl.*;

import java.util.*;

public class DataMerge extends PekkoStreamlet {
  AvroInlet<Data> inlet1 = AvroInlet.<Data>create("in-0", Data.class);
  AvroInlet<Data> inlet2 = AvroInlet.<Data>create("in-1", Data.class);
  AvroOutlet<Data> outlet = AvroOutlet.<Data>create("out",  d -> d.getKey(), Data.class);

  public StreamletShape shape() {
    return StreamletShape.createWithInlets(inlet1, inlet2).withOutlets(outlet);
  }

  public RunnableGraphStreamletLogic createLogic() {
    return new RunnableGraphStreamletLogic(getContext()) {
      public RunnableGraph<?> createRunnableGraph() {
        return Merger.source(getContext(), inlet1, inlet2).to(getCommittableSink(outlet));
      }
    };
  }
}
// end::merge[]

File: examples/snippets/modules/ROOT/examples/pekkostreams-scala/src/main/scala/cloudflow/pekkostreamsdoc/DataMerge.scala
package cloudflow.pekkostreamsdoc

// tag::merge[]
import cloudflow.pekkostream._
import cloudflow.pekkostream.scaladsl._
import cloudflow.pekkostream.util.scaladsl._
import cloudflow.streamlets.avro._
import cloudflow.streamlets._

class DataMerge extends PekkoStreamlet {

  val in0 = AvroInlet[Data]("in-0")
  val in1 = AvroInlet[Data]("in-1")
  val out = AvroOutlet[Data]("out", d => d.key)

  final override val shape = StreamletShape.withInlets(in0, in1).withOutlets(out)

  override final def createLogic = new RunnableGraphStreamletLogic {
    def runnableGraph =
      Merger.source(in0, in1).to(committableSink(out))
  }
}
// end::merge[]
